tools/optrack/find-latency-spikes.py

```diff
@@ -0,0 +1,1063 @@
+    global arrowRightImg;
+
+    previousLink = "";
+    nextLink = "";
+
+    # Strip the directory component out of the file name.
+    #
+    if previous is not None:
+        words = previous.split("/");
+        previousStripped = words[len(words)-1];
+        previousLink =  "<a href=\"" + previousStripped + "\">" + \
+                        "<img src=\"" + arrowLeftImg + \
+                        "\" height=\"30\" style=\"float:left\"></a><p>&nbsp;";
+
+
+    if next is not None:
+        words = next.split("/");
+        nextStripped = words[len(words)-1];
+        nextLink = "<a href=\"" + nextStripped + "\">" + \
+                   "<img src=\"" + arrowRightImg + \
+                   "\" height=\"30\" style=\"float:right\"></a><p>&nbsp;";
+
+    line = previousLink + " " + nextLink + "\n";
+    return line;
+
+
+# Into the current file insert links to the previous one and to the next one.
+# The rewritten file is saved under a new file name.
+#
+def linkFiles(current, previous, next):
+
+    curFile = None;
+    newFile = None;
+    newFileName = current + ".new";
+
+    try:
+        curFile = open(current, "r");
+    except:
+        print(color.RED + color.BOLD),
+        exc_type, exc_value, exc_traceback = sys.exc_info()
+        traceback.print_exception(exc_type, exc_value, exc_traceback);
+        print("Could not open file " + current + " for reading.");
+        print(color.END);
+        return None;
+
+    try:
+        newFile = open(newFileName, "w");
+    except:
+        print(color.RED + color.BOLD),
+        exc_type, exc_value, exc_traceback = sys.exc_info()
+        traceback.print_exception(exc_type, exc_value, exc_traceback);
+        print("Could not open file " + newFileName + " for writing.");
+        print(color.END);
+        return None;
+
+    curFileLines = curFile.readlines();
+
+    for i in range(len(curFileLines)):
+        line = curFileLines[i];
+
+        insertedLine = makeLineWithLinks(previous, next);
+
+        if "<body>" in line:
+            curFileLines.insert(i+1, insertedLine);
+        elif "</body>" in line:
+            curFileLines.insert(i, insertedLine);
+
+    for line in curFileLines:
+        newFile.write(line);
+
+    curFile.close();
+    newFile.close();
+
+    os.rename(newFileName, current);
+
+# We have a list of bucket files. Each one is an HTML file showing a slice of
+# the execution. To be able to easily navigate between consecutive execution
+# slices we insert links into each slice-file that take us to the previous
+# slice and to the next slice.
+#
+def interlinkFiles(fnameList):
+
+    for i in range(len(fnameList)):
+        current = fnameList[i];
+
+        if i > 0:
+            previous = fnameList[i-1];
+        else:
+            previous = None;
+
+        if (i < len(fnameList)-1):
+            next = fnameList[i+1];
+        else:
+            next = None;
+
+        linkFiles(current, previous, next);
+
+def processFile(fname):
```

Comment:
It might help understanding the flow of the program if processFile and the functions it calls were connected by name.  Maybe:
  `processFIle` => `inputFile`
  `createCallstackSeries` => `inputDoCallstacks`
  `getIntervalData` => `inputIntervalData`
etc. That would make clear that there's a bunch of things happening while input files are being processed, and other things happening after that.  Alternatively, the input side of things could be packaged in a class instance, which might help manage many of the global variables.